src/app/store/reducers/app.reducer.ts

```diff
@@ -57,6 +59,24 @@ export function reducer(
         pageTitle
       };
     }
+
+    case fromAction.SET_USER_ROLES: {
+      // TODO prehaps find better solution for rendering sequence of nav tabs. It will not work Fees Acc
+      const roles = [...action.payload].sort();
```

Comment:
i agree. This looks like it will cause the  tab order to change if the set of roles changes. I think this will work - instead of roles.forEach do the following (not tested or compiled, but hopefully you get the idea

 allNavItems.forEach (nav => {
   roles.some( role => {
     if (state.allNavItems.hasOwnProperty(role)) {
          navItems = [
            ...navItems,
            state.allNavItems[role]
          ];
     return true;
     } else {
      return false;
     }

Reply:
this is not going to work i am afraid. as we need to sort out the array in order to match our rendered tab sequence. I am not sure how to that elegantly. The reason why we need to do this is because the array of roles returned by sidam is not in the sequence we need. I base my navigation on those roles

Reply:
Sorting the roles doesn't seem to do what you want. Perhaps you can add an order member to the allNavItems collection?